frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretsList.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  ResourceLink,
+  Firehose,
+  FirehoseResult,
+  FirehoseResource,
+} from '@console/internal/components/utils';
+import { SecretModel, ServiceAccountModel } from '@console/internal/models';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecondaryStatus } from '@console/shared';
+import { SecretKind, ServiceAccountKind } from '@console/internal/module/k8s';
+import { PIPELINE_SERVICE_ACCOUNT } from '../const';
+import './SecretsList.scss';
+
+type SecretsProps = {
+  secrets?: FirehoseResult<SecretKind[]>;
+  serviceaccounts?: FirehoseResult<ServiceAccountKind>;
+};
+
+type SecretsListProps = {
+  namespace: string;
+};
+
+const secretTypes = [SecretType.dockerconfigjson, SecretType.basicAuth, SecretType.sshAuth];
+
+const Secrets: React.FC<SecretsProps> = ({ secrets, serviceaccounts }) => {
+  const servicAccountSecrets = _.map(serviceaccounts.data.secrets, 'name');
+  const filterData = _.filter(
+    secrets.data,
+    (secret) =>
+      _.includes(secretTypes, secret.type) &&
+      _.includes(servicAccountSecrets, secret.metadata.name),
+  );
+  const sortedFilterData = _.sortBy(filterData, (data) => data.metadata.name);
+
+  return (
+    <div className="odc-secrets-list">
+      <label>Secrets ({sortedFilterData.length})</label>
+      <div className="odc-secrets-list__secrets">
+        {sortedFilterData.map((secret) => {
+          return (
+            <ResourceLink
+              className="odc-secrets-list__secrets--item"
+              key={secret.metadata.uid}
+              kind={SecretModel.kind}
+              name={secret.metadata.name}
+              namespace={secret.metadata.namespace}
+              title={secret.metadata.name}
+            />
+          );
+        })}
+        {_.isEmpty(sortedFilterData) && <SecondaryStatus status="No source secrets found" />}
+      </div>
+    </div>
+  );
+};
+
+const SecretsList: React.FC<SecretsListProps> = ({ namespace }) => {
+  const resources: FirehoseResource[] = [
+    {
+      isList: true,
+      namespace,
+      kind: SecretModel.kind,
+      prop: SecretModel.plural,
+    },
+    {
+      isList: false,
+      namespace,
+      kind: ServiceAccountModel.kind,
+      prop: ServiceAccountModel.plural,
+      name: PIPELINE_SERVICE_ACCOUNT,
+    },
+  ];
+
+  return (
+    <Firehose resources={resources}>
```

Comment:
I'd like to see more usage of `Watchk8sResource` being used instead of `Firehose` it's far superior method to fetching items. But let's put that on the back burner... maybe add it to the list of cleanup stuff - log a task please.

Reply:
Issue to track : https://issues.redhat.com/browse/ODC-3550

Reply:
FWIW, I tried using it earlier while I was refactoring a bunch of helm stuff but it just didn't work for me. It went into infinite state updates. Maybe I was doing something wrong there but it led me to fallback to firehose.

Reply:
@rohitkrai03 Yup, you'll have that problem I suspect if you're trying to get namespaced resources and are creating the object inline.

`React.useMemo` is needed to make sure you don't recompute your resources on every render loop... otherwise you'll end up freshly saying "here are different resources" to the hook which naturally tosses everything out and starts again.

Can this be improved so it does a deep compare and we don't need useMemo? Probably. Might be worth mentioning it in #forum-ui.

Reply:
Yeah, I figured something of sorts was happening when I tried to use it but didn't wanted to spend much time on it. I definitely think use of `useMemo` should not be required and the hook should figure out that the dependency has not been changed automatically. Maybe we can make use of `useDeepCompareMemoize()` hook that I created somehow to move this logic into `useK8sWatchResource(s)` hooks.